packages/devtools_shared/lib/src/deeplink/deeplink_manager.dart

```diff
@@ -23,150 +24,140 @@ class DeeplinkManager {
   /// {"configurations":["Debug","Release","Profile"],"targets":["Runner","RunnerTests"]}
   static final _iosBuildOptionsJsonRegex = RegExp(r'({.*})');
 
+  /// The key to retrieve error message from the returning map of this class's
+  /// APIs.
   static const kErrorField = 'error';
+
+  /// The key to retrieve output json from the returning map of this class's
+  /// APIs.
   static const kOutputJsonField = 'json';
 
+  /// A regex to retrieve the file path from the stdout of iOS or Android
+  /// analyzers.
+  ///
+  /// Example stdout:
+  ///
+  /// result saved in /path/to/json/file.json
   static final _outputFilePathRegex = RegExp(r'result saved in (.*.json)');
 
   @visibleForTesting
   Future<ProcessResult> runProcess(
     String executable, {
     required List<String> arguments,
-    required String workingDirectory,
   }) {
     return Process.run(
       executable,
       arguments,
-      workingDirectory: workingDirectory,
     );
   }
 
-  Future<ProcessResult> _runFlutterCommand(List<String> arguments) {
-    final String flutterRoot = Platform.environment['FLUTTER_ROOT']!;
-    return runProcess(
-      'flutter',
-      arguments: arguments,
-      workingDirectory: flutterRoot,
+  @visibleForTesting
+  String? getFlutterBinary() {
+    final String? flutterRoot = Platform.environment['FLUTTER_ROOT'];
+    if (flutterRoot == null) {
+      return null;
+    }
+    return path.join(
+      flutterRoot,
+      'bin',
+      Platform.isWindows ? 'flutter.bat' : 'flutter',
     );
   }
 
-  Future<Map<String, String?>> _runFlutterCommandAndReadOutputFile(
-    List<String> arguments,
-  ) async {
-    final ProcessResult result = await _runFlutterCommand(arguments);
+  Future<String> _runFlutterCommand(
+    List<String> arguments, {
+    required RegExp outputMatcher,
+  }) async {
+    final flutterPath = getFlutterBinary();
+    if (flutterPath == null) {
+      throw 'Running Flutter command is not supported in this devtool version';
```

Comment:
Is it really the DevTools version that determines this? We should be more clear about which version is preventing the `FLUTTER_ROOT` environment variable from being set. In other words, is this due to the Dart SDK version, the Flutter SDK version, etc.

Reply:
Hi @DanTup Do you know which dart version the change will be in?

Reply:
ah I see, there isn't a version we can target in this case, a lot of different cases can result in this situation. For example running the dev tool directly through a standalone dart. I think I will just mention that it can't find a flutter sdk and maybe mention it needs to set FLUTTER_ROOT environment variable

Reply:
Can we just use 'flutter' if we don't find FLUTTER_ROOT? Not as accurate as using FLUTTER_ROOT but we would at least not fail as easily since most users should have `flutter` on their path

Reply:
> Hi @DanTup Do you know which dart version the change will be in?

Version 3.74.0 of the Dart extension for VS Code will include the change to set `FLUTTER_ROOT`. As you noted, this is specific to the server being started by VS Code though, so may not apply if it's run from the command line, etc. so if it's missing, it's worth trying to handle gracefully.

Perhaps before falling back to using `flutter` from `PATH` (which may also not exist, or be a different version if the user is using something like `fvm`), it's worth checking `Platform.resolvedExecutable` (which is the path to the Dart VM that's running) and walk up to see if it's nested inside `bin/cache/dart-sdk/bin` of a Flutter SDK and using that?

Reply:
I can try to use the flutter from path.

>  it's worth checking Platform.resolvedExecutable (which is the path to the Dart VM that's running) and walk up to see if it's nested inside bin/cache/dart-sdk/bin of a Flutter SDK and using that?

If it is using the dart from the flutter sdk, it should have set the FLUTTER_ROOT I think?

Reply:
> If it is using the dart from the flutter sdk, it should have set the FLUTTER_ROOT I think?

I _think_ if it's using the shell script at `bin/dart` it would, but if it was launched directly using `bin/cache/dart-sdk/bin/dart` I'm not sure it would. I don't know how common each way is though (personally I don't use the shell script because `dart` on my `PATH` is always a standalone Dart SDK, but my setup is likely not the same as most users :-))

Reply:
done